src/PBase/Utility/HexEncoding.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Text;
+
+namespace PBase.Utility
+{
+    public static class HexEncoding
+    {
+        /// <summary>Converts a text string into a hex encoded format.</summary>
+        /// <param name="stringToHex">The string to be hex encoded.</param>
+        /// <returns>The hex encoded string.</returns>
+        public static string ToHex(this string stringToHex)
+        {
+            if (string.IsNullOrEmpty(stringToHex))
+            {
+                throw new System.ArgumentException("String to hex cannot be null or empty");
+            }
+
+            var bytes = Encoding.Default.GetBytes(stringToHex);
+            var hexString = BitConverter.ToString(bytes);
+            hexString = hexString.Replace("-", "");
+            hexString = hexString.ToLower();
```

Comment:
Might want to allow people to supply arguments to say if this wish this to be set to lower and replace certain characters

Reply:
Added boolean arguments to handle whether we want to reformat the hex string in the encode method, defaulted to true so as to not change the behaviour. Added an extra step to remove the dashes in the decode method as well, in case the string provided has not been formatted the remove the dashes as per the default. Unsure of if I need to change the exception in the decode method regarding the invalid hex string length, as now hex strings can be odd numbered in length so long as the odd number is a result of the dashes, and not the hex code characters themselves.

Reply:
What if someone wants it as uppercase?

Reply:
I don't really think that should be the responsibility of the hex encoder. I would suggest that it should only perform the hex encoding and decoding, and if the user wants to format the response, then they can call ToUpper or ToLower on the resulting string themselves. If anything I'd remove the toLower (and potentially even the removeDashes arguments), and have it return the same each time.

Reply:
Lets go for that then